api/api_test.go

```diff
@@ -1,21 +1,24 @@
-package api
+package api_test
 
 import (
 	"context"
-	"github.com/gorilla/mux"
 	"net/http/httptest"
 	"testing"
 
+	"github.com/ONSdigital/dp-upload-service/api"
+	"github.com/ONSdigital/dp-upload-service/api/mock"
+	"github.com/gorilla/mux"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestSetup(t *testing.T) {
 	Convey("Given an API instance", t, func() {
-		r := mux.NewRouter()
+		r := &mux.Router{}
 		ctx := context.Background()
-		api := Setup(ctx, r)
+		api := api.Setup(ctx, &mock.VaultClienterMock{}, r, &mock.S3ClienterMock{})
```

Comment:
```suggestion
		apiMock := Setup(ctx, &mock.VaultClienterMock{}, r, &mock.S3ClienterMock{})
```

You can call the Setup function without the need for `api.Setup` if the package name is `api`.
Also, to clearly distinguish one from another, you can name the variable as `apiMock` or `MockAPI`